### PR TITLE
Refine:  make multi skeleton component shared one instance of spine.Skeleton while spine working at sharedCache mode

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #
https://github.com/cocos/cocos-engine-external/pull/449
https://github.com/cocos/cocos-engine/issues/16598

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/editor-support/spine-wasm/spine-skeleton-instance.cpp [hunk 1]
```diff
@@ -577,4 +578,17 @@ void SpineSkeletonInstance::setSlotTexture(const std::string &slotName, uint32_t
...
+
+void SpineSkeletonInstance::addRef() {
+    ++_referenceCount;
+}
+
+void SpineSkeletonInstance::release() {
+    --_referenceCount;
+
+    if (_referenceCount == 0) {
+        SpineSkeletonSystem::removeSpineInstance(this);
```
Comment (minggo) [inline native/cocos/editor-support/spine-wasm/spine-skeleton-instance.cpp]: It is better to do it in destructor.

File: cocos/spine/skeleton-cache.ts [hunk 2]
```diff
@@ -92,15 +92,23 @@ export class AnimationCache {
     public totalTime = 0;
     public frames: AnimationFrame[] = [];
 
-    constructor (data: spine.SkeletonData) {
+    constructor (data: spine.SkeletonData, animCache: AnimationCache | null) {
```
Comment (minggo) [inline cocos/spine/skeleton-cache.ts]: It is better not to add a variable here and determine its validation in codes.

File: native/cocos/editor-support/spine-wasm/spine-skeleton-instance.cpp [hunk 3]
```diff
@@ -577,4 +578,16 @@ void SpineSkeletonInstance::setSlotTexture(const std::string &slotName, uint32_t
...
         slotTextureSet[slot] = textureID;
     }
+}
+
+void SpineSkeletonInstance::addRef() {
+    ++_referenceCount;
+}
+
+void SpineSkeletonInstance::release() {
+    --_referenceCount;
```
Comment (minggo) [inline native/cocos/editor-support/spine-wasm/spine-skeleton-instance.cpp]: It is better to assert _referenceCount > 0.

File: cocos/spine/skeleton-cache.ts [hunk 4]
```diff
@@ -338,7 +342,28 @@ class SkeletonCache {
...
+                return;
+            }
+            this._sharedCacheMap.delete(uuid);
+        }
+
+        const sharedOperate = (aniKey: string, animationCache: AnimationCache): void => {
+            this._animationPool[`${uuid}#${aniKey}`] = animationCache;
+            animationCache.clear();
+        };
+        const privateOperate = (aniKey: string, animationCache: AnimationCache): void => {
```
Comment (dumganhar) [inline cocos/spine/skeleton-cache.ts:363]: Why is it called `privateOperate`? What does `private` mean here?

File: cocos/spine/skeleton-cache.ts [hunk 5]
```diff
@@ -347,17 +372,29 @@ class SkeletonCache {
...
+                this._sharedCacheMap.set(uuid, sharedInstances);
+            }
+            if (sharedInstances.indexOf(skeletonComponent) === -1) {
+                sharedInstances.push(skeletonComponent);
+            }
+        }
         let skeletonInfo = this._skeletonCache[uuid];
         if (!skeletonInfo) {
-            const skeleton = null;
+            const skeleton = new spine.Skeleton(skeletonData);
```
Comment (dumganhar) [inline cocos/spine/skeleton-cache.ts]: skeleton is forever `null` in the old code?

File: cocos/spine/skeleton-cache.ts (hunk 4 above)
Comment (bofeng-song) [reply cocos/spine/skeleton-cache.ts:363]: private corresponds to privateCache mode.

File: cocos/spine/skeleton-cache.ts (hunk 5 above)
Comment (bofeng-song) [reply cocos/spine/skeleton-cache.ts]: skeleton will constructor by skeleton component's _instance in old code

Comment (dumganhar) [reply cocos/spine/skeleton-cache.ts]: So we move the `spine.Skeleton` construction code to here?

**Review by dumganhar — APPROVED** [review]

File: cocos/spine/skeleton-cache.ts (hunk 5 above)
Comment (bofeng-song) [reply cocos/spine/skeleton-cache.ts]: Yes, previous logic multi skeleton component can not share the same object of spine.skeleton

**Review by minggo — APPROVED** [review]

**Pull request merged**